openedx/core/djangoapps/user_api/profiles/views.py

```diff
@@ -0,0 +1,64 @@
+"""
+NOTE: this API is WIP and has not yet been approved. Do not use this API without talking to Christina or Andy.
+
+For more information, see:
+https://openedx.atlassian.net/wiki/display/TNL/User+API
+"""
+from django.conf import settings
+from django.contrib.auth.models import User
+
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework.authentication import OAuth2Authentication, SessionAuthentication
+from rest_framework import permissions
+
+from ..accounts.views import AccountView
+from ..models import UserPreference
+
+from . import PROFILE_VISIBILITY_PREF_KEY
+
+class ProfileView(APIView):
+    """
+        **Use Cases**
+
+            Get the user's public profile information.
```

Comment:
I find the "public profile" confusing. If the user is calling it, he can get all his information. If it is a different user, it will return only the public information (if user has made rest of profile private).